frontend/src/app/plugins/file.cljs

```diff
@@ -105,7 +111,47 @@
       :else
       (let [page-id (uuid/next)]
         (st/emit! (dw/create-page {:page-id page-id :file-id $id}))
-        (page/page-proxy $plugin $id page-id)))))
+        (page/page-proxy $plugin $id page-id))))
+
+  (export
+    [self type export-type]
+    (let [export-type (or (parser/parse-keyword export-type) :all)]
+      (cond
+        (not (contains? #{"penpot" "zip"} type))
+        (u/display-not-valid :export-type type)
+
+        (not (contains? (set mue/export-types) export-type))
+        (u/display-not-valid :export-exportType export-type)
+
+        :else
+        (let [export-cmd (if (= type "penpot") :export-binary-file :export-standard-file)
+              file (u/proxy->file self)
+              features (features/get-team-enabled-features @st/state)
+              team-id  (:current-team-id @st/state)]
+          (p/create
+           (fn [resolve reject]
+             (->> (uw/ask-many!
+                   {:cmd export-cmd
+                    :team-id team-id
+                    :features features
+                    :export-type export-type
+                    :files [file]})
+                  (rx/mapcat #(->> (rx/of %) (rx/delay 1000)))
+                  (rx/mapcat
+                   (fn [msg]
+                     (case (:type msg)
+                       :error
+                       (rx/throw "Cannot export file")
```

Comment:
Consider throwing an instance of js/Error or ex-info which is already an instance of js/Error instead of text. Throwing text is not recommended